runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobInvoker.java

```diff
@@ -66,7 +67,9 @@ protected JobInvocation invokeWithExecutor(
         String.format("%s_%s", flinkOptions.getJobName(), UUID.randomUUID().toString());
 
     if (FlinkPipelineOptions.AUTO.equals(flinkOptions.getFlinkMaster())) {
-      flinkOptions.setFlinkMaster(serverConfig.getFlinkMaster());
+      // Although Flink uses Rest, it expects the address not to contain a http scheme
+      String flinkMaster = stripHttpSchema(serverConfig.getFlinkMaster());
```

Comment:
I'm not sure why we should strip the schema in this particular place?

Reply:
What is a better place?

Reply:
My understanding is that since you're stripping the schema in `FlinkExecutionEnvironments`, it shouldn't be necessary to also do it here.

Reply:
True, that is redundant.